src/Features/Core/Portable/UseCollectionInitializer/ObjectCreationExpressionAnalyzer.cs

```diff
@@ -118,10 +118,22 @@ protected override void AddMatches(ArrayBuilder<TExpressionStatementSyntax> matc
             }
         }
 
-        protected override bool ShouldAnalize()
-            => _semanticModel.GetTypeInfo(_objectCreationExpression, default).Type
-                    .GetMembers(WellKnownMemberNames.CollectionInitializerAddMethodName)
-                    .Any(x => x is IMethodSymbol m && m.DeclaredAccessibility == Accessibility.Public);
+        protected override bool ShouldAnalyze()
+        {
+            var type = _semanticModel.GetTypeInfo(_objectCreationExpression, _cancellationToken).Type;
```

Comment:
:memo: This should not cause a performance problem because this expression was already being used on this this code path:

https://github.com/dotnet/roslyn/blob/84e1af7365956a04027014ce48c228f874a93d3c/src/Features/Core/Portable/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs#L88-L89